### PR TITLE
Stop masking non-modular packages that only provide masked name

### PR DESCRIPTION
**Pull request by lubomir**
If a default module contains package foo, and there's a nonmodular foo1, which provides foo, we don't want to mask the non-modular package. It doesn't conflict with the module in any way.

### CODE REVIEW

File: fus.c [hunk 1]
```diff
@@ -670,10 +677,22 @@ mask_solvable_bare_rpms (Pool *pool)
...
+              if (modpkg->name == pkg->name)
+                {
+                  g_debug ("Excluding %s (masked by %s in %s)",
+                          pool_solvable2str (pool, pkg),
+                          pool_solvable2str (pool, modpkg),
+                          pool_solvable2str (pool, s));
+                  selection_make (pool,
+                          &selection,
+                          pool_solvid2str (pool, *mp),
+                          SELECTION_CANON | SELECTION_ADD);
```
Comment (lubomir) [inline fus.c:693]: There's probably a better way to make this query. Essentially what's needed is to find solvables with name identical to `modpkg->name` that do not provide `modular-package()`.

Comment (igor-raits) [reply fus.c:693]: on RPM level you could do `(name WITHOUT modular-package())` but I won't find time to look into this code ever…

**Comment by ppisar** [comment]
While this proposed change looks like a fix, it diverges from a DNF behavior which is documented in dnf.modularity(7).

**Comment by lubomir** [comment]
Yeah, you're right. This is the relevant part:

> the non-modular packages are filtered out when their name or provide matches against  a       modular  package  name  from any enabled, default, or dependent stream.

**Pull request closed without merging**